src/python/gudhi/representations/vector_methods.py

```diff
@@ -350,6 +352,154 @@ def __call__(self, diag):
         """
         return self.fit_transform([diag])[0,:]
 
+
+class BettiCurve2(BaseEstimator, TransformerMixin):
+    """
+    A more flexible replacement for the BettiCurve class.
```

Comment:
Doc is a bit short, but the exact doc we want depends on if we merge this with the other BettiCurve or not. Are infinite values supported? (I think so)
It took me a bit of time to understand what the example was saying, maybe a sentence or two that says the same thing in English instead of code would help understand it (the example itself is good).

Reply:
Indeed. I'll expand the docs a bit.

Re infinites: Infinite deaths are supported. Infinite Betti numbers are not. I should check that the implementation doesn't go nuts if fed infinite births.